java-checks/src/main/java/org/sonar/java/symexecengine/ExecutionState.java

```diff
@@ -142,6 +146,20 @@ private void reportIssuesToTopState(Set<Tree> trees) {
     return results;
   }
 
+  public boolean hasState(Symbol symbol, Class<? extends State> stateClass) {
+    Set<Value> values = reachableValues.get(symbol);
+    if(values.isEmpty()) {
+      return false;
+    }
+    for (Value value : values) {
+      State state = getStateOfValue(value);
+      if(state!= null && !state.getClass().equals(stateClass)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
   // FIXME : Hideous hack for closeable to get "Ignored" variables
```

Comment:
![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Take the required action to fix the issue indicated by this comment.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)